ibm-sai.cpp

```diff
@@ -0,0 +1,70 @@
+#include "ibm-sai.hpp"
+
+#include "utils.hpp"
+
+#include <phosphor-logging/lg2.hpp>
+
+namespace phosphor
+{
+namespace led
+{
+namespace ibm
+{
+// Set OperationalStatus functional according to the asserted state of the group
+void setOperationalStatus(const std::string& path, bool value)
+{
+    if (path != PARTITION_SAI || path != PLATFORM_SAI)
+    {
+        return;
+    }
+
+    // Get endpoints from the rType
+    std::string fru = path + "/fault_led_group";
+
+    // endpoint contains the vector of strings, where each string is a Inventory
+    // D-Bus object that this, associated with this LED Group D-Bus object
+    // pointed to by fru_fault
+    utils::PropertyValue endpoint{};
+
+    try
+    {
+        endpoint = utils::DBusHandler().getProperty(
+            fru, "xyz.openbmc_project.Association", "endpoints");
+    }
+    catch (const sdbusplus::exception::SdBusError& e)
+    {
+        lg2::error(
+            "Failed to get endpoints property, ERROR = {ERROR}, PATH = {PATH}",
+            "ERROR", e.what(), "PATH", fru);
+        return;
+    }
+
+    auto& endpoints = std::get<std::vector<std::string>>(endpoint);
+    if (endpoints.empty())
+    {
+        return;
+    }
+
+    for (const auto& fruInstancePath : endpoints)
+    {
+        // Set OperationalStatus by fru instance path
+        try
+        {
+            utils::PropertyValue functionalValue{value};
+            utils::DBusHandler().setProperty(
+                fruInstancePath,
+                "xyz.openbmc_project.State.Decorator.OperationalStatus",
+                "Functional", functionalValue);
```

Comment:
The use case here is actually slightly different - We have two LED group objects, the PLATFORM_SAI and PARTITION_SAI pointing to the same inventory object (See https://github.com/openbmc/openbmc/blob/a2cc401d586a9b1a3cdf2e93847839b551b48074/meta-ibm/recipes-phosphor/inventory/phosphor-inventory-manager/p10bmc/ibm%2Ceverest_associations.json#L293).

So when setting Operational status to true, we need to sure that both groups are de-asserted.

If we are setting OperationalStatus to true, it is OK to just look at the asserted for the incoming path.